Some small tweaks/suggestions for the method section (#49)
BIDS-numpy/numpy-paper: methods.tex

The \code{ndarray} is the data structure at the heart of NumPy.
The data structure stores regularly strided homogeneous data types

> inside a contiguous block memory, allowing for the efficient representation

Author: not sure it makes a difference, I stumbled a bit because a sliced array itself is not contiguous anymore, of course it still  is a "view" into a contiguous block.

Reviewer: Good point!

the ufunc will take care of looping over the dimensions of the inputs
in such a way that if, say, \code{x} is a 1-D array with length 3, and
\code{y} is a 2-D array with shape $2 \times 1$, the output will be

> an array with shape $2 \times 3$ (Fig.~\ref{fig:array-concepts}c).

Author: I assume it is OK to reference images in the main article?

The ufunc machinery takes care
of calling the function with all the appropriate combinations of
input array elements to complete the output array.
The elementary arithmetic operations of addition, multiplication, etc.,

> are implemented as ufuncs, so broadcasting applies also to expressions

Author: ```suggestion
are implemented as ufuncs, so broadcasting applies even in expressions
```
stumbled over "even", but rereading "also" is too weak...

Reviewer: How about "so that broadcasting applies also to"?

Author: Sounds good. The even does not really fit here I think, but it had the nice ring of being deeply ingrained into everything numpy does.

distributions, and more.

\paragraph{\emph{Infrastructure libraries.}} NumPy provides utilities
for writing tests and for building Python packages.

The \code{testing} subpackage provides functions such as
\code{assert\_allclose(actual, desired)} that may be used in
test suites for code that uses NumPy arrays.

NumPy provides the subpackage \code{distutils} which includes functions and classes
to facilitate configuration, installation, and packaging of libraries depending on NumPy.

> These can be used, for example, to as part of publishing to the PyPI website.

Author: I assume we were talking about distutils here? I tried to split things up a bit, not sure it was fully successfull.

Reviewer: "to as part" -> ...

Although, I'm not sure our customizations of distutils have much to do with PyPi?

NumPy adopted an official Governance Document on October~5,
2015\cite{NumPyProjectGovernance}.
Project decisions are usually made by consensus of interested contributors.

> This means for most decisions everyone is entrusted with veto power.

Reviewer: This means that, for most decisions, everyone is...

Author: If this feels too strong, we could add the word "initially", or similar.

accelerated linear algebra.

Finally, since May 2019 the project receives around 10K USD annually from

> % typically is correct right now, but do we want to write it so specifically?

Author: Was just curious if we want to keep it so specific (it is fully correct now of course). I am happy with it as is though.

Reviewer: "a small amount" may suffice; it's a small enough amount that it probably doesn't matter either way :)

Reviewer 2: Let's go with small amount.

wraps the core pseudorandom number generator based on a Mersenne Twister
implementation\footnote{to be precise, the standard 32-bit version of MT19937};
interfaces with the underlying code that transforms random bits into

> variates from other distributions; and supplies a singleton instance exposed in

Author: I stumbled over "deviates" a bit, probably just me, but variates was used all over.

Reviewer: deviates should probably used throughout; not sure how that slipped in there!

Reviewer 2: I think "variates" is correct, no?

> A random variate is a particular outcome of a random variable: the random variates which are other outcomes of the same random variable might have different values. A random deviate or simply deviate is the difference of random variate with respect to the distribution central location (e.g., mean), often divided by the standard deviation of the distribution.

- https://en.wikipedia.org/wiki/Random_variate

from a system-dependent provider, or with a user-provided seed. The seed
sequence then transforms the initial set of entropy into a sequence of
high-quality pseudorandom integers, which can be used to initialize multiple bit

> generators deterministically. A key design element of a seed sequence is that

Reviewer: a key feature of a seed sequence, perhaps?

> it can be used to spawn child \code{SeedSequence}s which are distinct from

Reviewer: What does distinct mean here?  Do you think we need to be more precise, or is that enough?

Author: Good one, just realized, we also really need to cite Professor O'Neils work here if we did not already. She probably has some nice phrase for distinct, although the opposite direction to just tell "its safe to use it to create as many generators as you wish" may be just as well.
